sdcm/sct_config.py

```diff
@@ -1161,6 +1212,11 @@ def _load_environment_variables(self):
                     raise ValueError("failed to parse {} from environment variable:\n\t\t{}".format(opt['env'], ex))
         return environment_vars
 
+    def _update_environment_variables(self, replace=False):
```

Comment:
what that ? why are we update all the variables into the os.environ ? we have them available via params almost across the whole SCT code ?

Reply:
:) Yes, this looks little bit, hmm, riotish, but idea behind this is providing parameters to shell commands which run from the SCT Python code. We had a discussion with @bentsi about this stuff and found that it'll be handy.

I used it to substitute parameters in .yaml files using `envsubst` command without any temporary files.

Reply:
O.K. using `envsubst` is a nice tricky, but...

If we ever reenter those functions again (maybe if we'll ever run multiple tests in one go), I fear it might break some of the parameters. or leave leftover to the next test.

Also the ones that are a bit more complex structure, like a dict of a list, maybe break again.

I'll might limit that to specific parameters, and tag them in the params list...

Reply:
None of that problems exists, from my pov:

1. Re-entering the function is no-op effectively. You can force environment update by calling with `replace=True`, but it's not a default behavior.
1. All environment updates done for the current process and it's childs only, so there is no way to break the next test.
1. I reviewed all parameters and I'm sure that for all of them serialization using `str()` works fine (we use 8 option types: `float`, `int`, `boolean`, `str_or_list`, `list`, `bool`, `int_or_list`.)
1. I'm in doubt about "secret" parameters only, but we have no such parameters right now. If they'll appear in the future, I prefer to have a blocklist instead.